vulnfeeds/cmd/download-cves/main.go

```diff
@@ -74,26 +83,64 @@ func downloadCVE2(apiKey string, cvePath string) {
 	if err != nil {
 		log.Fatalf("Error reading response body: %+v", err)
 	}
-	var NVD2Data cves.NVDCVE2
+	var NVD2Data *cves.NVDCVE2
 	err = json.Unmarshal(body, &NVD2Data)
 	if err != nil {
 		log.Fatalf("Failed to decode NVD data: %+v", err)
 	}
-	Logger.Infof("%d results to download", NVD2Data.TotalResults)
-	// TODO: figure out how to paginate
-	file.Write(body)
+	Logger.Infof("At offset %d of total %d results", *NVD2Data.StartIndex, *NVD2Data.TotalResults)
+	return *NVD2Data
+}
+
+// Download all of the CVE data using the 2.0 API
+// See https://nvd.nist.gov/developers/vulnerabilities
+func downloadCVE2(APIKey string, CVEPath string) {
+	file, err := os.OpenFile(path.Join(CVEPath, "nvdcve-2.0.json.new"), os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0644)
+	defer file.Close()
+	if err != nil { // There's an existing file, check if it matches server file
+		Logger.Fatalf("Something went wrong when creating/opening file: %+v", err)
+	}
+	completeNVDdataset := cves.NVDCVE2{}
+	page := cves.NVDCVE2{}
+	offset := 0
+	for {
+		page = downloadCVE2WithOffset(APIKey, offset)
+		if len(page.Vulnerabilities) == 0 {
```

Comment:
We can save a request here by placing the if break after 112, and checking to see if `offset` is > than the `page.TotalResults`

Reply:
Thanks. Fixed in https://github.com/google/osv.dev/pull/773/commits/1ccbd66e02c6bbb955918765d67e6c1c264c4afd and https://github.com/google/osv.dev/pull/773/commits/28fdb2ce9316469e1ef1d850ac4f372c8519d638

Reply:
Actually, this is now going to break prematurely, it's not going to run for the final partial page of results, right?

Reply:
No, never mind, I'm not thinking straight.